data-prepper-plugins/http-source-common/src/test/java/org/opensearch/dataprepper/http/codec/JsonCodecTest.java

```diff
@@ -69,20 +70,29 @@ public void testParseSuccessWithMaxSize() throws IOException {
     @ParameterizedTest
     @ArgumentsSource(JsonArrayWithKnownFirstArgumentsProvider.class)
     public void parse_should_return_lists_smaller_than_provided_length(
-            final String inputJsonArray, final String knownFirstPart, int expectedChunks) throws IOException {
+            final String inputJsonArray, final String knownFirstPart, final int maxSize, final List<List<String>> expectedChunks, final List<Boolean> exceedsMaxSize) throws IOException {
         final int knownSingleBodySize = knownFirstPart.getBytes(Charset.defaultCharset()).length;
-        final int maxSize = (knownSingleBodySize * 2) + 3;
+        //final int maxSize = (knownSingleBodySize * 2) + 3;
         final List<List<String>> chunkedBodies = objectUnderTest.parse(HttpData.ofUtf8(inputJsonArray),
                 maxSize);
 
         assertThat(chunkedBodies, notNullValue());
-        assertThat(chunkedBodies.size(), equalTo(expectedChunks));
-        final String firstReconstructed = chunkedBodies.get(0).stream().collect(Collectors.joining(",", "[", "]"));
-        assertThat(firstReconstructed.getBytes(Charset.defaultCharset()).length,
-                lessThanOrEqualTo(maxSize));
-
-        assertThat(chunkedBodies.get(0).size(), greaterThanOrEqualTo(1));
-        assertThat(chunkedBodies.get(0).get(0), equalTo(knownFirstPart));
+        assertThat(chunkedBodies.size(), equalTo(expectedChunks.size()));
+
+        for (int i = 0; i < expectedChunks.size(); i++) {
+            final String reconstructed = chunkedBodies.get(i).stream().collect(Collectors.joining(",", "[", "]"));
+            if (exceedsMaxSize.get(i)) {
```

Comment:
Might it be simpler to provide the expected size and then do a single `assertThat(..., equalTo(expectedSize))`?

Reply:
I didn't want to be so precise.